packages/terra-navigation-side-menu/src/NavigationSideMenu.jsx

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import List from 'terra-list';
+import ActionHeader from 'terra-clinical-action-header';
+import ContentContainer from 'terra-content-container';
+import RoutingStackDelegate from 'terra-navigation-layout/lib/RoutingStackDelegate';
+
+import styles from './NavigationSideMenu.scss';
+
+const cx = classNames.bind(styles);
+
+const KEYCODES = {
+  ENTER: 13,
+};
+
+const propTypes = {
+  /**
+   * An array of configuration for each menu item.
+   */
+  menuItems: PropTypes.arrayOf(PropTypes.shape({
+    /**
+     * Keys of menu items
+     */
+    children: PropTypes.arrayOf(PropTypes.string),
+    /**
+     * ID to be applied to the menu item div.
+     */
+    id: PropTypes.string,
+    /**
+     * Unique identifier that will be returned in the onChange callback when an endpoint is reached.
+     */
+    key: PropTypes.string.isRequired,
+    /**
+     * Title for the menu row and header title when selected.
+     */
+    title: PropTypes.string.isRequired,
```

Comment:
Should this prop be named `text` to align with [application-links](https://github.com/cerner/terra-framework/blob/master/packages/terra-application-links/src/list/ApplicationList.jsx#L18)?